docs/reference/run.md

```diff
@@ -970,59 +970,31 @@ $ docker run -it --entrypoint="" mysql bash
 
 ### Exposed ports
 
-The following `run` command options work with container networking:
-
-    --expose=[]: Expose a port or a range of ports inside the container.
-                 These are additional to those exposed by the `EXPOSE` instruction
-    -P         : Publish all exposed ports to the host interfaces
-    -p=[]      : Publish a container's port or a range of ports to the host
-                   format: ip:hostPort:containerPort | ip::containerPort | hostPort:containerPort | containerPort
-                   Both hostPort and containerPort can be specified as a
-                   range of ports. When specifying ranges for both, the
-                   number of container ports in the range must match the
-                   number of host ports in the range, for example:
-                       -p 1234-1236:1234-1236/tcp
-
-                   When specifying a range for hostPort only, the
-                   containerPort must not be a range.  In this case the
-                   container port is published somewhere within the
-                   specified hostPort range. (e.g., `-p 1234-1236:1234/tcp`)
-
-                   (use 'docker port' to see the actual mapping)
-
-    --link=""  : Add link to another container (<name or id>:alias or <name or id>)
-
-With the exception of the `EXPOSE` directive, an image developer hasn't
-got much control over networking. The `EXPOSE` instruction defines the
-initial incoming ports that provide services. These ports are available
-to processes inside the container. An operator can use the `--expose`
-option to add to the exposed ports.
-
-To expose a container's internal port, an operator can start the
-container with the `-P` or `-p` flag. The exposed port is accessible on
-the host and the ports are available to any client that can reach the
-host.
-
-The `-P` option publishes all the ports to the host interfaces. Docker
-binds each exposed port to a random port on the host. The range of
-ports are within an *ephemeral port range* defined by
-`/proc/sys/net/ipv4/ip_local_port_range`. Use the `-p` flag to
-explicitly map a single port or range of ports.
```

Comment:
Do we still document this information elsewhere? the `/proc/sys/net/ipv4/ip_local_port_range` information may be relevant information.

Reply:
IMO, covering this is going into too much detail. If you really want to know about how the high ports are assigned, you can see the source code for the port allocator.

Reply:
Do we have this documented elsewhere? I know it's been a frequent topic for various setups with customers, and we should document how to prevent docker from conflicting with other tools (which may mean re-configuring `/proc/sys/net/ipv4/ip_local_port_rang`

Reply:
OK; I added a `--publish-all` section in the CLI reference page, which included the same bits as highlighted here, plus this addition about `ip_local_port_rang`